Creating a data model

VicPeralta opened the pull request:
# Creating a data model

In this PR:

- No linter errors
- Correct GitFlow used
- Repo documented professionally
- Created migration files to create tables users, posts, comments, likes
- Table and columns names follow naming conventions
- Foreign keys included
- Indexes created on foreign key columns

![image](https://user-images.githubusercontent.com/10048317/171530501-f6de0596-fb52-4ab9-9522-a25ed7463a91.png)

VicPeralta committed:
Commands to create tables using rails g migration

new file:   commandsDatabase.md

VicPeralta committed:
Update commandsDatabase to include indexes

modified:   commandsDatabase.md

VicPeralta committed:
Add migration files

new file:   db/migrate/20220601235024_create_users.rb
new file:   db/migrate/20220601235048_create_posts.rb
new file:   db/migrate/20220602000603_create_comments.rb
new file:   db/migrate/20220602001050_create_likes.rb
new file:   db/schema.rb

hiromataba commented on db/migrate/20220601235024_create_users.rb:
```diff
@@ -0,0 +1,12 @@
+    create_table :users do |t|
+      t.string :name
+      t.string :photo
+      t.string :bio
```
- [x] The bio should be with type of `text` because it may be longer please update that

VicPeralta replied:
Right, thanks

hiromataba commented on db/migrate/20220601235048_create_posts.rb:
```diff
@@ -0,0 +1,14 @@
+    create_table :posts do |t|
+      t.integer :user_id
+      t.string :title
+      t.string :text
```
- [x] The text should be with the type of `text` because it may be longer please update that

VicPeralta replied:
Thanks

hiromataba reviewed (changes requested):
Hi @VicPeralta     :wave: :wave: 

![almost-there](https://user-images.githubusercontent.com/75126481/160092588-b3f37e8e-a1d9-4b14-84fc-7ec1fed7b3e5.gif)

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Your PR has a proper title and description  :+1: 
- Your linters are passing :heavy_check_mark:  
- Profesional README file :rocket: 

## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

VicPeralta committed:
Changes based on code review

Change t:string to t:text on Users.Bio and Posts.text

modified:   commandsDatabase.md
modified:   db/migrate/20220601235024_create_users.rb
modified:   db/migrate/20220601235048_create_posts.rb

ahmetbozaci commented on db/migrate/20220602000603_create_comments.rb:
```diff
@@ -0,0 +1,15 @@
+    create_table :comments do |t|
+      t.integer :user_id
+      t.integer :post_id
+      t.string :text
```
### OPTIONAL
_This is a requirement not mentioned in the last review so please fix it._

- [x] Comment's text type should be also text. Please change it.

VicPeralta replied:
Thanks, I changed it.

ahmetbozaci reviewed (approved):
# STATUS: APPROVED 🥳 🥳 🥳 
![](https://media.giphy.com/media/3pZipqyo1sqHDfJGtz/giphy.gif)

@VicPeralta  ✋🏼 
#### Your project is great. 🎉 I approve of it. There is nothing else to say other than... it's time to merge it. You can submit this project on your dashboard.

___ 

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- [x] user_id should be an author_id according to the schema. Please change it.

- [x]  ![image](https://user-images.githubusercontent.com/38635158/171639196-aacd736e-afdd-4cd7-80af-a2df7643c66f.png)

Your tables should be like that.
![image](https://user-images.githubusercontent.com/38635158/171640523-2ecd308b-7461-4dd1-bb63-54de7e9f71f8.png)

___

### Cheers and Happy coding! 💻 👏
![aproved](https://media.giphy.com/media/xT0xezQGU5xCDJuCPe/giphy.gif)


___

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification._
___

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_

[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

VicPeralta committed:
Change name of column user_id to author_id on Posts, Likes and Comments table

Create Index on post
Change data type from string to text on Comments.text

modified:   commandsDatabase.md
modified:   db/migrate/20220601235048_create_posts.rb
modified:   db/migrate/20220602000603_create_comments.rb
modified:   db/migrate/20220602001050_create_likes.rb
modified:   db/schema.rb
